src/sonic-yang-models/yang-models/sonic-buffer-pool.yang

```diff
@@ -0,0 +1,63 @@
+module sonic-buffer-pool {
+    namespace "http://github.com/Azure/sonic-buffer-pool";
+    prefix bpl;
+
+    organization
+        "SONiC";
+
+    contact
+        "SONiC";
+
+    description
+        "SONIC BUFFER POOL";
+
+    revision 2021-07-01 {
+        description
+            "Initial revision.";
+    }
+
+    container sonic-buffer-pool {
+
+        container BUFFER_POOL {
+
+            list BUFFER_POOL_LIST {
+                key "name";
+
+                leaf name {
+                    type string;
+                    description "Buffer Pool name";
```

Comment:
I'm not sure we allow any name here, I see the backend code written with a specific name e.g egress_lossless_pool, ingress_lossless_pool..etc, please check and restrict the name usage.

Reply:
Hi Venkat
   egress_lossless_pool/ingress_lossy_pool etc are defined in buffer_defaults j2 files only. There is no restriction for the name in the cpp files and it is just a string and it just has to be unique only. I dont think there is a need of name restriction here. 
Thanks
Amit

Reply:
Please see the code below, there is an assumption to use a particular pool name.
https://github.com/Azure/sonic-swss/blob/master/cfgmgr/buffermgr.cpp#L173

Reply:
This is added only for ingress_lossless_pool in sonic-swss/cfgmgr/buffermgr.h.
#define INGRESS_LOSSLESS_PG_POOL_NAME "ingress_lossless_pool"
What about other buffer pools- egress/ingress lossy/lossless. (egress_lossy_pool/ingress_lossy_pool) etc?

Reply:
ok, let's have a string type for now to accept any pool name.